src/Symfony/Component/EventDispatcher/DependencyInjection/RegisterListenersPass.php

```diff
@@ -182,6 +167,130 @@ private function getEventFromTypeDeclaration(ContainerBuilder $container, string
+    {
+        if (!isset($eventDefinition['event'])) {
+            $eventDefinition['method'] ??= '__invoke';
+            $eventDefinition['event'] = $this->getEventFromTypeDeclaration($container, $serviceId, $eventDefinition['method']);
+        }
+
+        $eventDefinition['event'] = $aliases[$eventDefinition['event']] ?? $eventDefinition['event'];
+
+        if (!isset($eventDefinition['method'])) {
+            $eventDefinition['method'] = 'on'.preg_replace_callback([
+                    '/(?<=\b|_)[a-z]/i',
+                    '/[^a-z0-9]/i',
+                ], fn ($matches) => strtoupper($matches[0]), $eventDefinition['event']);
+            $eventDefinition['method'] = preg_replace('/[^a-z0-9]/i', '', $eventDefinition['method']);
+
+            if (null !== ($class = $container->getDefinition($serviceId)->getClass()) && ($r = $container->getReflectionClass($class, false)) && !$r->hasMethod($eventDefinition['method']) && $r->hasMethod('__invoke')) {
+                $eventDefinition['method'] = '__invoke';
+            }
+        }
+    }
+
+    private function normalizeBeforeAfter(string|null $beforeAfter, array $aliases): string|null
+    {
+        if (null === $beforeAfter) {
+            return null;
+        }
+
+        $beforeAfter = explode('::', $beforeAfter);
+
+        return sprintf(
+            '%s::%s',
+            $aliases[$beforeAfter[0]] ?? $beforeAfter[0],
+            trim($beforeAfter[1] ?? '__invoke', '()')
+        );
+    }
+
+    private function computePriority(array $listeners, string $listenerName, array $alreadyVisited = []): int
+    {
+        if ($alreadyVisited[$listenerName] ?? false) {
+            throw new InvalidArgumentException(sprintf('Circular reference detected for listener "%s".', $listenerName));
+        }
+
+        $alreadyVisited[$listenerName] = true;
+
+        if (isset($listeners[$listenerName]['before'])) {
+            return $this->computePriority($listeners, $listeners[$listenerName]['before'], $alreadyVisited) + 1;
+        }
+
+        if (isset($listeners[$listenerName]['after'])) {
+            return $this->computePriority($listeners, $listeners[$listenerName]['after'], $alreadyVisited) - 1;
+        }
+
+        return $listeners[$listenerName]['priority'] ?? 0;
+    }
+
+    private function updateListenerPriority(ContainerBuilder $container, string $eventName, string $listenerName, int $priority): void
+    {
+        $listenerDefinition = $container->getDefinition($listenerName);
+        $events = $listenerDefinition->getTag('kernel.event_listener');
+
+        foreach ($events as &$event) {
+            if ($event['event'] === $eventName) {
+                $event['priority'] = $priority;
+                break;
+            }
+        }
+
+        $listenerDefinition->clearTag('kernel.event_listener');
+        $listenerDefinition->addTag('kernel.event_listener', $events);
```

Comment:
By doing that, every `kernel.event_listener` tag will have the same priority, no matter the event, isn't that an issue?

Reply:
either I don't understand your point, or I think you're wrong: 
see previous loop, and notice the condition:
```php
        foreach ($events as &$event) {
            if ($event['event'] === $eventName) {
                $event['priority'] = $priority;
                break;
            }
        }
```

AFAIK there is no way to update only ONE event/listenero definition for a given service

Reply:
Indeed, I was wrong 😅